content/en/docs/Reference/stage_modules.md

```diff
@@ -0,0 +1,449 @@
+- **primary-group**: Default group for the user. Defaults to a new group created named after the user.
+- **groups**: Add user to these additional groups. Kairos creates an `admin` group by default which is also added to the sudoers file. Add a user to the `admin` group if you want them to have `sudo` access.
+- **no-user-group**: Boolean. Skip default group creation.
+- **ssh-authorized-keys**: List of public SSH keys to authorize for this user
+- **system**: Create the user as a system user. No home directory will be created.
+- **no-log-init**: Boolean. Skip initialization of lastlog and faillog databases.
+- **shell**: User's login shell.
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup users"
+       users: 
+          bastion: 
+            passwd: "strongpassword"
+            homedir: "/home/foo
+```
+
+### `authorized_keys`
+
+A list of SSH authorized keys that should be added for each user.
+SSH keys can be obtained from GitHub user accounts by using the format github:${USERNAME}, similarly for GitLab with gitlab:${USERNAME}.
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup exception trace"
+       authorized_keys:
+         mudler:
+         - "github:mudler"
+         - "ssh-rsa: ..."
+```
+
+### `modules`
+
+A list of kernel modules to load.
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup users"
+       modules:
+       - nvidia
+```
+
+### `timesyncd`
+
+Sets the `systemd-timesyncd` daemon file (`/etc/system/timesyncd.conf`) file accordingly. The documentation for `timesyncd` and all the options can be found [here](https://www.freedesktop.org/software/systemd/man/timesyncd.conf.html).
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup NTP"
+       systemctl:
+         enable:
+         - systemd-timesyncd
+       timesyncd:
+          NTP: "0.pool.org foo.pool.org"
+          FallbackNTP: ""
+          ...
+```
+
+### `systemctl`
+
+A list of systemd services to `enable`, `disable`, `mask` or `start`.
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup users"
+       systemctl:
+         enable:
+          - systemd-timesyncd
+          - cronie
+         mask:
+          - purge-kernels
+         disable:
+          - crond
+         start:
+          - cronie
+```
+
+### `environment_file`
+
+A string to specify where to set the environment file
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup users"
+       environment_file: "/home/user/.envrc"
+       environment:
+         FOO: "bar"
+```
+
+### `environment`
+
+A map of variables to write in `/etc/environment`, or otherwise specified in `environment_file`
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup users"
+       environment:
+         FOO: "bar"
+```
+
+### `systemd_firstboot`
+
+Runs [`systemd-firstboot`](https://www.freedesktop.org/software/systemd/man/latest/systemd-firstboot.html) with the arguments specified.
+
+```yaml
```

Comment:
I tried this example and it doesn't seem to have any effect. If I run `systemd-firstboot --hostname=myhostname` manually it creates `/etc/hostname` correctly. I can see in the agent logs that the stage was run but for some reason it had no effect.